apps/design/backend/src/test_decks.test.ts

```diff
@@ -0,0 +1,136 @@
+import {
+  electionGeneralDefinition,
+  electionTwoPartyPrimaryDefinition,
+} from '@votingworks/fixtures';
+import {
+  BallotLayout,
+  DEFAULT_LAYOUT_OPTIONS,
+  layOutAllBallotStyles,
+} from '@votingworks/hmpb-layout';
+import {
+  BallotType,
+  ElectionDefinition,
+  getContests,
+} from '@votingworks/types';
+import { assert, iter } from '@votingworks/basics';
+import {
+  getBallotStyleById,
+  getBallotStylesByPrecinctId,
+  numBallotPositions,
+} from '@votingworks/utils';
+import { createPrecinctTestDeck } from './test_decks';
+
+function expectedTestDeckPages(
+  ballots: BallotLayout[],
+  electionDefinition: ElectionDefinition
+): number {
+  return iter(ballots)
+    .map((ballot) => {
+      const { document, gridLayout } = ballot;
+      const ballotStyle = getBallotStyleById(
+        electionDefinition,
+        gridLayout.ballotStyleId
+      );
+      const contests = getContests({
+        election: electionDefinition.election,
+        ballotStyle,
+      });
+      const maxContestOptions = Math.max(...contests.map(numBallotPositions));
```

Comment:
While a finished election will presumably always have contest in each ballot style, it's possible that an in-progress election definition will not. In that case `contests` would be `[]` which means `maxContestOptions` will be `-Infinity`, causing the final result of `expectedTestDeckPages` to also be `-Infinity`. This is just a test helper, so it's not a huge deal as we're not dealing with arbitrary user input. However, I'd still suggest this which makes the "empty contests" handling more explicit:
```suggestion
      const maxContestOptions =
        iter(contests).map(numBallotPositions).max() ?? 0;
```

Reply:
Great catch, fixed here: 2a6da512e